uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/WordApp.java

```diff
@@ -135,6 +134,18 @@ private void signIn(@NonNull final String username,
     @Override
     public void confirmAccount(@NonNull final String username) {
         Logger.i(TAG, "Confirming account with supplied username is signed in..");
+
+        // Had a screen for microsoft 365 pop up occasionally
+        final UiObject msft365Object = UiAutomatorUtils.obtainUiObjectWithText("Go Premium with Microsoft 365 Personal");
+        if (msft365Object.exists()) {
```

Comment:
Do we know how much this waits for existence?

Reply:
Using UiAutomatorUtils and calling the obtain methods calls a waitForExists as long as FIND_UI_ELEMENT_TIMEOUT. I think here, that would be 25 seconds.